crates/foundry/evm/core/src/fork/multi.rs

```diff
@@ -122,19 +122,24 @@ impl MultiFork {
     /// Returns a fork backend
     ///
     /// If no matching fork backend exists it will be created
-    pub fn create_fork(&self, fork: CreateFork) -> eyre::Result<(ForkId, SharedBackend, Env)> {
+    pub fn create_fork(
+        &self,
+        fork: CreateFork,
+    ) -> eyre::Result<(ForkId, SharedBackend, Env, Option<u64>)> {
+        let fork_block_number = fork.evm_opts.fork_block_number;
         trace!(
             "Creating new fork, url={}, block={:?}",
             fork.url,
-            fork.evm_opts.fork_block_number
+            fork_block_number
         );
         let (sender, rx) = oneshot_channel();
         let req = Request::CreateFork(Box::new(fork), sender);
         self.handler
             .clone()
             .try_send(req)
             .map_err(|e| eyre::eyre!("{:?}", e))?;
-        rx.recv()?
+        let (fork_id, shared_backend, env) = rx.recv()??;
+        Ok((fork_id, shared_backend, env, fork_block_number))
```

Comment:
Why return `fork_block_number` from this function when it's already part of the input? Can't the caller just re-use the input value?

Reply:
Good catch thanks! Fixed in https://github.com/NomicFoundation/edr/pull/822/commits/74a47d17aa564b3169657b43397285f2ad62c31e